source/github/github.go

```diff
@@ -43,27 +44,29 @@ type Config struct {
 }
 
 func (g *Github) Open(url string) (source.Driver, error) {
+	var client *http.Client
 	u, err := nurl.Parse(url)
 	if err != nil {
 		return nil, err
 	}
 
-	if u.User == nil {
-		return nil, ErrNoUserInfo
-	}
-
-	password, ok := u.User.Password()
-	if !ok {
-		return nil, ErrNoUserInfo
-	}
+	// client defaults to http.DefaultClient
+	client = nil
```

Comment:
Replace this with `var client *http.Client`. You don't need to set the pointer to `nil` since it'll default to nil.

Reply:
Nice catch. Do you think `// client defaults to http.DefaultClient` comment is needed above `var client *http.Client`?

Reply:
Yeah, let's leave that comment